cpp/test/Ice/optional/Test.ice

```diff
@@ -119,19 +119,12 @@ class A
     optional(500) int mc;
 }
 
-["preserve-slice"]
 class B extends A
 {
     int requiredB = 0;
     optional(10) int md;
 }
 
-class C extends B
```

Comment:
I would keep C like in the AMD version.

Reply:
I removed it because it didn't seem very useful to test slicing here. But on second thoughts, it's indeed useful to ensure the "has optional members" information isn't lost for the unknown slice. I've added it back.